src/components/fileUpload.js

```diff
@@ -322,12 +498,129 @@
+        display: 'flex',
+      },
+      gridItem: {
+        display: 'flex',
+        borderRadius: '0.3rem',
+        flexDirection: 'column',
+        border: '1px solid #eee',
+        marginRight: '0.9375rem',
+        marginBottom: '0.9375rem',
+      },
+      gridItemDetails: {
+        maxWidth: ({ options: { imagePreviewWidth, showImagePreview } }) =>
+          showImagePreview ? imagePreviewWidth : 'auto',
+        display: 'flex',
+        margin: '0.9375rem',
+        justifyContent: 'space-between',
+      },
+      devImage: {
+        display: 'flex',
+        overflow: 'hidden',
+        margin: '0.9375rem',
+        borderStyle: 'dashed',
+        borderColor: '#AFB5C8',
+        borderRadius: '0.3rem',
+        borderWidth: '0.0625rem',
+        border: '2px dashed black',
+        backgroundColor: '#F0F1F5',
+        width: ({ options: { imagePreviewWidth } }) => imagePreviewWidth,
+        height: ({ options: { imagePreviewHeight } }) => imagePreviewHeight,
+      },
+      gridDevImage: {
+        extend: 'devImage',
+
+        margin: 0,
+        borderRadius: '0.3rem 0.3rem 0 0',
+        width: ({ options: { imagePreviewWidth } }) => imagePreviewWidth,
+        height: ({ options: { imagePreviewHeight } }) => imagePreviewHeight,
+      },
+      deleteIcon: {
+        color: `${t.colors.light}!important`,
+      },
+      uploadingImage: {
+        border: 'none',
+        color: 'white',
+        display: 'flex',
+        overflow: 'hidden',
+        margin: '0.9375rem',
+        alignItems: 'center',
+        borderRadius: '0.3rem',
+        justifyContent: 'center',
+        backgroundColor: t.colors.warning,
+        width: ({ options: { imagePreviewWidth } }) => imagePreviewWidth,
+        height: ({ options: { imagePreviewHeight } }) => imagePreviewHeight,
+      },
+      gridUploadingImage: {
+        extend: 'uploadingImage',
+
+        margin: 0,
+        borderRadius: '0.3rem 0.3rem 0 0',
+        width: ({ options: { imagePreviewWidth } }) => imagePreviewWidth,
+        height: ({ options: { imagePreviewHeight } }) => imagePreviewHeight,
+      },
+      image: {
+        overflow: 'hidden',
+        margin: '0.9375rem',
+        borderRadius: '0.3rem',
+        backgroundSize: 'cover',
+        width: ({ options: { imagePreviewWidth } }) => imagePreviewWidth,
+        height: ({ options: { imagePreviewHeight } }) => imagePreviewHeight,
+      },
+      gridImage: {
+        margin: 0,
+        overflow: 'hidden',
+        backgroundSize: 'cover',
+        borderRadius: '0.3rem 0.3rem 0 0',
+        width: ({ options: { imagePreviewWidth } }) => imagePreviewWidth,
+        height: ({ options: { imagePreviewHeight } }) => imagePreviewHeight,
+      },
+      fileDetails: {
+        flexGrow: 1,
+        maxWidth: ({
+          options: { type, imagePreviewWidth, showImagePreview },
+        }) =>
+          showImagePreview && type === 'grid'
+            ? `calc(${imagePreviewWidth} - 58px)`
```

Comment:
Question: Do you need the `calc`? What type is `imagePreviewWidth`? A string or a number?

Reply:
Good one. It's a string, but I'm able to parse it via js: `${parseInt(imagePreviewWidth, 10) - 58}px`, which I believe would be nicer.

Reply:
Hmm if it is a string I guess the unit is in there as well? `${parseInt(imagePreviewWidth, 10) - 58}px` won't work if the `imagePreviewWidth` is for example `1rem`. Your original solution will work.

Reply:
Good point. I've reverted it back to `calc`.